src/app/project/ontology/ontology.service.ts

```diff
@@ -148,20 +143,23 @@ export class OntologyService {
                             i.guiEle === property.guiElement && i.subPropOf === subProp
                         ));
                     } else {
-
-                        // if the property is type of number or list, the gui element is not relevant
-                        // because the app supports only one gui element (at the moment): the spinbox resp. the list pulldown
-                        if (property.objectType === Constants.DecimalValue || property.objectType === Constants.ListValue) {
+                        if (property.objectType === Constants.IntValue && subProp === Constants.SeqNum) {
+                            // if the property is of type number, but sub property of SeqNum,
+                            // select the correct default prop params
                             propType = (group.elements.find(i =>
-                                i.objectType === property.objectType
+                                i.objectType === property.objectType && i.subPropOf === Constants.SeqNum
                             ));
-                        } else if (property.objectType === Constants.IntValue && subProp === Constants.SeqNum) {
+                        } else if (property.objectType === Constants.TextValue) {
+                            // if the property is of type text value, we have to check the gui element
+                            // to get the correct default prop params
                             propType = (group.elements.find(i =>
-                                i.objectType === property.objectType && i.subPropOf === Constants.SeqNum
+                                i.guiEle === property.guiElement && i.objectType === property.objectType
                             ));
                         } else {
+                            // in all other cases the gui-element resp. the subProp is not relevant
+                            // because the object type is unique
```

Comment:
Small observation: In line 139/140 you put the comment before the _if_ structure and here after the _if_ and _else_

Reply:
Good point. It's always the question, where to put the comment. Normally we put it outside the if statement. But in this long list of `if`, `else if`, `else` I thought it makes sense to have it inside.